Exercise/VSIX/BrowseCommand.cs

```
@@ -44,8 +47,18 @@ private BrowseCommand(AsyncPackage package, OleMenuCommandService commandService
 
             var menuCommandID = new CommandID(CommandSet, CommandId);
             
-            var menuItem = new MenuCommand(this.Execute, menuCommandID);
+            var menuItem = new MenuCommand(this.BrowseFile, menuCommandID);
             commandService.AddCommand(menuItem);
+
+            try
+            {
+                var comp = this.package.GetService<SComponentModel, IComponentModel>();
+                configProvider = comp.GetService<IConfigProvider>();
+            }
+            catch (Exception exception)
+            {
+                Debug.WriteLine(exception);
```

Squashing the exception here will cause a crash when we try to use the `null` `configProvider` in line 109.
How could we avoid the crash and improve the UX?

---

hmm, I mean im not quite sure what you mean? If it cannot find the mef component than shouldnt it crash?

---

No, but the `configProvider` will be `null`. If user clicks on the command and selects a file, we'll end up calling `configProvider.UpdateConfiguration(...)` and get a null-reference exception.